flocker/acceptance/test_api.py

```diff
@@ -109,3 +110,101 @@ def got_body(body):
+                         b"", None, True)
+        self.addCleanup(close, agent1)
+
+        agent2 = run_SSH(22, 'root', node_2,
+                         [b"flocker-zfs-agent", node_2, node_1],
+                         b"", None, True)
+        self.addCleanup(close, agent2)
+
+        api_starting = wait_for_api(node_1)
+        uuid = unicode(uuid4())
+        dataset = {u"primary": node_1,
+                   u"dataset_id": uuid,
+                   u"metadata": {u"name": u"my_volume"}}
+        base_url = b"http://{}:{}/v1".format(node_1, REST_API_PORT)
+        dataset_requested = api_starting.addCallback(
+            lambda _: post(base_url + b"/configuration/datasets",
+                           data=dumps(dataset),
+                           headers={b"content-type": b"application/json"},
+                           persistent=False))
+        dataset_requested.addCallback(content)
+
+        def got_result(result):
+            result = loads(result)
+            self.assertEqual(dataset, result)
+        dataset_acknowledged = dataset_requested.addCallback(got_result)
+
+        def created():
+            result = get(base_url + b"/state/datasets", persistent=False)
+            result.addCallback(content)
+
+            def got_body(body):
+                body = loads(body)
+                # Current state listing includes bogus metadata
+                # https://clusterhq.atlassian.net/browse/FLOC-1386
+                expected_dataset = dataset.copy()
+                expected_dataset[u"metadata"].clear()
+                return expected_dataset in body
+            result.addCallback(got_body)
+            return result
+        dataset_created = dataset_acknowledged.addCallback(lambda _: loop_until(created))
+        def move(ignored):
+            moved_dataset = {
+                u'primary': node_2
+            }
+            post(base_url + b"/configuration/datasets/%s" % (dataset['dataset_id'].encode('ascii')),
+                           data=dumps(moved_dataset),
+                           headers={b"content-type": b"application/json"},
+                           persistent=False)
+        dataset_moving = dataset_created.addCallback(move)
+
+        def moved():
+            result = get(base_url + b"/state/datasets", persistent=False)
+            result.addCallback(content)
+
+            def got_body(body):
+                body = loads(body)
+                # Current state listing includes bogus metadata
+                # https://clusterhq.atlassian.net/browse/FLOC-1386
+                expected_dataset = dataset.copy()
+                expected_dataset[u'primary'] = node_2
+                expected_dataset[u"metadata"].clear()
+                return expected_dataset in body
+            result.addCallback(got_body)
+            return result
+        dataset_moved = dataset_moving.addCallback(lambda _: loop_until(moved))
+
+        return dataset_moved
```

Comment:
A few PEP8 complaints about this file: http://build.clusterhq.com/builders/flocker-lint/builds/3879/steps/lint/logs/stdio